src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,965 @@
+                    x_scan_delay=self.parent._ebeam_scanner.scanDelay.value[0],
+                    y_prescan_lines=self.parent._ebeam_scanner.scanDelay.value[1],
+                    flyback_time=int(self.parent._ebeam_scanner.scanFlyback.value /
+                                     self.parent._ebeam_scanner.clockPeriod.value),
+                    x_scan_offset=self.parent._ebeam_scanner.scanOffset.value[0],
+                    y_scan_offset=self.parent._ebeam_scanner.scanOffset.value[1],
+                    x_scan_gain=self.parent._ebeam_scanner.scanGain.value[0],
+                    y_scan_gain=self.parent._ebeam_scanner.scanGain.value[1],
+                    x_descan_gain=self.parent._mirror_descanner.scanGain.value[0],
+                    y_descan_gain=self.parent._mirror_descanner.scanGain.value[1],
+                    x_descan_offset=self.parent._mirror_descanner.scanOffset.value[0],
+                    y_descan_offset=self.parent._mirror_descanner.scanOffset.value[1],
+                    scan_rotation=self.parent._ebeam_scanner.rotation.value,
+                    descan_rotation=self.parent._mirror_descanner.rotation.value,
+            )
+
+        return megafield_metadata
+
+    def _acquire(self):
+        """
+        Acquisition thread takes input from the self.acq_queue which holds a command ('start', 'next', 'stop',
+        'terminate') and extra arguments (MegaFieldMetaData Model or FieldMetaData Model and the notifier function to
+        which any return will be redirected)
+        """
+
+        try:
+            acquisition_in_progress = None  # To prevent that acquisitions mix up, or stop the acquisition twice.
+            tnext = 0
+
+            while True:
+                # Wait until a message is available
+                command, *args = self.acq_queue.get(block=True)
+
+                if command == "start":
+                    if acquisition_in_progress:
+                        logging.warning("ASM acquisition was already at status '%s'" % command)
+                        continue
+
+                    acquisition_in_progress = True
+                    megafield_metadata = args[0]
+                    self.parent.ASM_API_Post_Call("/scan/start_mega_field", 204, megafield_metadata.to_dict())
+
+                elif command == "next":
+                    if not acquisition_in_progress:
+                        logging.warning("Start ASM acquisition before taking field images")
+                        continue
+
+                    field_data = args[0]  # Field metadata for the specific position of the field to scan
+                    dataContent = args[1]  # Specifies the type of image to return (empty, thumbnail or full)
+                    notifier_func = args[2]  # Return function (usually, Dataflow.notify or acquire_single_filed queue)
+
+                    self.parent.ASM_API_Post_Call("/scan/scan_field", 204, field_data.to_dict())
```

Comment:
Is this a blocking call? If so, please indicate it. That's the trick to know that the acquisition is complete. Note that it'd be good if we could interrupt it with a stop or terminate event. Otherwise, if for some reason the dwell time is mistakenly too big, or the resolution too large, the acquisition might take a long time and be unstoppable. That said, you should first check that the hardware supports it.

Reply:
@pieleric  I assume this call i actually blocking. The timeout (defined in the call method) should prevent it from blocking the entire acquisition process. But I don't know for sure if it is possible to quit the acquisition in a call. I will ask Robert if this is possible.

Reply:
TODO  add the question if this call can be interupted to email which I will send to Robert.

Reply:
To cancel an acquisition, you can simply do not call "next" anymore, but "finish_acquisition" in case an cancellation event was received. That would mean we still keep the data acquired until that point, which I think is ok. That would also work if the dwell time is mistakenly too big or the resolution is too big. What do you think?

Reply:
That works indeed, but if the dwelltime is so long (and the connection bad etc.), the acquisition of 1 field image may take longer than the time out (10mintues). I don't no if before the time out is reached it is possible to cancel acquisition.
But this is a very corner case.

Reply:
The maximum dwelltime is 100us so we should never reach the limit of 10min.

Reply:
After discusioon --> TODO K.K. Implement a cancel method which clears the entire queue